internal/api/types_operation.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2025 Microsoft Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package api
+
+import (
+	"strings"
+	"time"
+
+	azcorearm "github.com/Azure/azure-sdk-for-go/sdk/azcore/arm"
+	"github.com/Azure/azure-sdk-for-go/sdk/data/azcosmos"
+
+	"github.com/Azure/ARO-HCP/internal/api/arm"
+)
+
+type OperationRequest string
+
+const (
+	OperationRequestCreate OperationRequest = "Create"
+	OperationRequestUpdate OperationRequest = "Update"
+	OperationRequestDelete OperationRequest = "Delete"
+
+	// These are for POST actions on resources.
+	OperationRequestRequestCredential OperationRequest = "RequestCredential"
+	OperationRequestRevokeCredentials OperationRequest = "RevokeCredentials"
+)
+
+type Operation struct {
+	CosmosUID string `json:"cosmosUID,omitempty"`
+
+	// ResourceID must be serialized exactly here for the generic CRUD to work.
+	// ResourceID here is NOT an ARM resourceID, it just parses like and one and is guarantee unique
+	ResourceID *azcorearm.ResourceID `json:"resourceId"`
+
+	// TenantID is the tenant ID of the client that requested the operation
+	TenantID string `json:"tenantId,omitempty"`
+	// ClientID is the object ID of the client that requested the operation
+	ClientID string `json:"clientId,omitempty"`
+	// Request is the type of asynchronous operation requested
+	Request OperationRequest `json:"request,omitempty"`
+	// ExternalID is the Azure resource ID of the cluster or node pool
+	ExternalID *azcorearm.ResourceID `json:"externalId,omitempty"`
```

Comment:
Do I want to know why this (and some other fields) are `omitempty`? May they be omitted? When?